### ISSUE TITLE
Failed to handle method call : java.lang.IllegalArgumentException: Failed to find configured root that contains /data/data/com.xxxx.yyyyy/app_flutter/Example_English.pdf

### ISSUE THREAD

**Issue by osamasado** [sidlatau/flutter_email_sender#66]
flutter_email_sender: ^4.0.0
path_provider: ^2.0.1

Please help me, how can I solve this issue?

**Comment by CodingWithTashi** [comment]
I had the same issue, I end up using getTemporaryDirectory from  path_provider instead of getApplicationDocumentsDirectory.

**Comment by CodingWithTashi** [comment]
Hi @LeapingSwan, You mentioned you are using `getApplicationDocumentsDirectory ` in your case and I can also see you are passing the path in PdfPreviewScreen. I not able to replicate your issue since code attached is just a snippet of your application. 
But what I could recomment you is that try `getTemporaryDirectory ` instead of `getApplicationDocumentsDirectory `.   
1. Here is how I am creating file (.txt). You might be having existing pdf etc   
```
/// This all field and method are define outside class so that you can use anywhere in your application

late File logFile;  // global file it can be txt,pdf,docx etc

Future<String> get _localPath async {
  final directory = await getTemporaryDirectory();
  return directory.path;
}

Future<File> get localFile async {
  final path = await _localPath;
  logFile = File('$path/logger.txt');
  logFile.writeAsString("==================LOG START FROM HERE=====================");
  return logFile;
}
```   
2. Here is how you can get the path and attach to mail   
```
 static void launchEmail(BuildContext context) async {
    try {
      final Email email = Email(
        body: 'Attaching log for ${getFormattedDateFromDateTime(
          dateTime: DateTime.now(),
        )}',
        subject: 'Log',
        recipients: ['abc@cde.com'], 
        attachmentPaths: [logFile.path],  // global file define at first
        isHTML: false,
      );
      await FlutterEmailSender.send(email);
    } on PlatformException catch (e) {
      ApplicationUtil.showSnackBar(context: context, message: e.message!);
    } catch (e) {
      print("=========================ERROR=========================");
      ApplicationUtil.showSnackBar(context: context, message: e.toString());
    }
  }
```   
I have tested in both IOS and android. It worked in both.

**Comment by LeapingSwan** [comment]
Ok I solved it using, as you suggested the getTemporaryDirectory, and the key to getting that working was putting the file in global location and querying it for it's own path when I needed it rather than trying to pass the path through.  That's brilliant, thanks so much for your help.

Stephen

**Comment by clebemachado** [comment]
Interesting, passing globally works. Worth the tip.

**Comment by sidlatau** [comment]
Added additional path for PathProvider:
```xml
<files-path name="files" path="../"/>
```
This seems to solve the access problem when the file is in `getApplicationDocumentsDirectory()`.

Fix will be available in v5.1.0.

I am considering adding:
```xml
<root-path name="root" path="." />
```
That also fixes a problem but it is not officially documented, so maybe the first solution will be enough.
